### PR TITLE
Added Boundary Signal Event

### LINKED ISSUE #1059: Add Signal Boundary Event
Interrupting signal boundary event:
Screen Shot 2020-01-27 at 2 28 42 PM

Non-interrupting signal boundary event:
Screen Shot 2020-01-27 at 2 28 53 PM

Acceptance criteria:
 * The boundary event can be attached to any task type.
 * The boundary event should have a triangle symbol like in the screenshots above.
   * There should be a solid double border if it's interrupting, and
   * a dashed double border if it's non-interrupting.
 * The signal boundary event should produce the following BPMN XML:
   ```
   <boundaryEvent id="boundary" attachedToRef="task" cancelActivity="true">
     <signalEventDefinition />
   </boundaryEvent>
   ```
   * If the boundary event is interrupting, omit `cancelActivity="true"` from the BPMN XML.
   * `attachedToRef` should be the ID of the task the boundary event is attached to.
 * The inspector config for the boundary event should have a Configuration accordion with a Name text input and an Interrupting checkbox:
   Screen Shot 2020-01-27 at 2 37 03 PM
 * The inspector config should also include an Advanced accordion with a Node Identifier text input:
   Screen Shot 2020-01-27 at 2 38 47 PM

Icon
[triangle.svg.zip](https://github.com/ProcessMaker/modeler/files/4119011/triangle.svg.zip)

### PR DESCRIPTION
**Pull request by lisa-fehr**
Closes #1059

### CODE REVIEW

File: tests/e2e/specs/BoundarySignalEvent.spec.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    const signalXML = '<bpmn:boundaryEvent id="node_3" name="Test name" attachedToRef="node_2"><bpmn:signalEventDefinition /></bpmn:boundaryEvent>';
+
+    cy.get('[data-test=downloadXMLBtn]').click();
+
+    cy.window()
+      .its('xml')
+      .then(removeIndentationAndLinebreaks)
+      .then(xml => {
+        expect(xml).to.contain(signalXML);
+      });
```
Comment (chipit24) [inline tests/e2e/specs/BoundarySignalEvent.spec.js]: Could you use the `assertDownloadedXmlContainsExpected` util here?

File: tests/e2e/specs/BoundarySignalEvent.spec.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    setBoundaryEvent(nodeTypes.boundarySignalEvent, taskPosition);
+  });
+
+  it('update boundary signal event properties element', () => {
+    moveElement(taskPosition, boundarySignalEventPosition.x, boundarySignalEventPosition.y);
+
+    const name = 'Test name';
+    typeIntoTextInput('[name=name]', name);
+
+    const signalXML = '<bpmn:boundaryEvent id="node_3" name="Test name" attachedToRef="node_2"><bpmn:signalEventDefinition /></bpmn:boundaryEvent>';
```
Comment (chipit24) [inline tests/e2e/specs/BoundarySignalEvent.spec.js]: You can re-use the `name` variable here:
```suggestion
    const signalXML = `<bpmn:boundaryEvent id="node_3" name="${name}" attachedToRef="node_2"><bpmn:signalEventDefinition /></bpmn:boundaryEvent>`;
```

File: tests/e2e/specs/BoundarySignalEvent.spec.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  const taskPosition = { x: 200, y: 200 };
+  const boundarySignalEventPosition = { x: 260, y: 200 };
+
+  beforeEach(() => {
+    dragFromSourceToDest(nodeTypes.task, taskPosition);
+    setBoundaryEvent(nodeTypes.boundarySignalEvent, taskPosition);
+  });
+
+  it('update boundary signal event properties element', () => {
+    moveElement(taskPosition, boundarySignalEventPosition.x, boundarySignalEventPosition.y);
```
Comment (chipit24) [inline tests/e2e/specs/BoundarySignalEvent.spec.js]: What's the reason for moving the shape like this? It doesn't look like you make any assertions against the task's position.

Comment (lisa-fehr) [reply tests/e2e/specs/BoundarySignalEvent.spec.js]: copy pasta

**Pull request merged**